warehouse/query-core/src/main/java/datawave/query/jexl/lookups/FieldNameIndexLookup.java

```diff
@@ -135,13 +145,18 @@ protected Callable<Boolean> createTimedCallable(final Iterator<Entry<Key,Value>>
 
         return () -> {
             try {
-                lookupStartTimeMillis = System.currentTimeMillis();
+                lookupStartTimeMillis.set(System.currentTimeMillis());
                 lookupStartedLatch.countDown();
 
                 final Text holder = new Text();
 
                 try {
                     while (iter.hasNext()) {
+                        // check for interrupt which may be triggered by closing the batch scanner
+                        if (Thread.currentThread().isInterrupted()) {
```

Comment:
Hmm.... we really need this check?  How can we be running if we were interrupted?

Reply:
without this check the thread does not reliably interrupt

Reply:
interesting, ok